dandiapi/api/models/upload.py

```diff
@@ -41,13 +41,13 @@ class Meta:
 
     @staticmethod
     @abstractmethod
-    def object_key(upload_id, dandiset: Dandiset | None = None):  # noqa: N805
```

Comment:
Note this code change slightly changes the way that `BaseUpload.object_key()` is called. Since `EmbargoedUpload` doesn't require `dandiset`, but `Upload` does, I changed `dandiset` to be a mandatory keyword argument.